src/net/codec.rs

```diff
@@ -329,78 +329,107 @@ impl StacksMessageCodec for Preamble {
+    fn consensus_deserialize<R: Read>(fd: &mut R) -> Result<GetBlocksInv, net_error> {
+        let consensus_hash: ConsensusHash             = read_next(fd)?;
+        let num_blocks : u16                          = read_next(fd)?;
+        if (num_blocks as u32) > BLOCKS_INV_DATA_MAX_BITLEN {
             // requested too long of a range 
-            return Err(net_error::DeserializeError(format!("Block diff is too big for inv ({} - {})", burn_height_start, burn_height_end)));
+            return Err(net_error::DeserializeError(format!("Block diff is too big for inv ({})", num_blocks)));
         }
 
-        Ok(GetBlocksData {
-            burn_height_start,
-            burn_header_hash_start,
-            burn_height_end,
-            burn_header_hash_end
-        })
-    }
-}
-
-impl StacksMessageCodec for MicroblocksInvData {
-    fn consensus_serialize<W: Write>(&self, fd: &mut W) -> Result<(), net_error> {
-        write_next(fd, &self.last_microblock_hash)?;
-        write_next(fd, &self.last_sequence)?;
-        Ok(())
-    }
-
-    fn consensus_deserialize<R: Read>(fd: &mut R) -> Result<MicroblocksInvData, net_error> {
-        let last_microblock_hash : BlockHeaderHash = read_next(fd)?;
-        let last_sequence : u16 = read_next(fd)?;
-
-        Ok(MicroblocksInvData {
-            last_microblock_hash,
-            last_sequence
+        Ok(GetBlocksInv {
+            consensus_hash: consensus_hash,
+            num_blocks: num_blocks
         })
     }
 }
 
 impl StacksMessageCodec for BlocksInvData {
     fn consensus_serialize<W: Write>(&self, fd: &mut W) -> Result<(), net_error> {
         write_next(fd, &self.bitlen)?;
-        write_next(fd, &self.bitvec)?;
-        write_next(fd, &self.microblocks_inventory)?;
+        write_next(fd, &self.block_bitvec)?;
+        write_next(fd, &self.microblocks_bitvec)?;
         Ok(())
     }
 
     fn consensus_deserialize<R: Read>(fd: &mut R) -> Result<BlocksInvData, net_error> {
-        let bitlen : u16                                     = read_next(fd)?;
+        let bitlen : u16 = read_next(fd)?;
         if bitlen > BLOCKS_INV_DATA_MAX_BITLEN as u16 {
             return Err(net_error::DeserializeError(format!("bitlen is bigger than max bitlen inv ({})", bitlen)));
         }
 
-        let bitvec : Vec<u8>                                = read_next_exact::<_, u8>(fd, BITVEC_LEN!(bitlen))?;
-        let microblocks_inventory : Vec<MicroblocksInvData> = read_next_exact::<_, MicroblocksInvData>(fd, bitlen as u32)?;
+        let block_bitvec : Vec<u8> = read_next_exact::<_, u8>(fd, BITVEC_LEN!(bitlen))?;
+        let microblocks_bitvec : Vec<u8> = read_next_exact::<_, u8>(fd, BITVEC_LEN!(bitlen))?;
 
         Ok(BlocksInvData {
             bitlen,
-            bitvec,
-            microblocks_inventory
+            block_bitvec,
+            microblocks_bitvec
         })
     }
 }
 
+impl BlocksInvData {
```

Comment:
Can this be deduped with net::inv::PeerBlockInv?

Reply:
The methods `has_ith_block` and `has_ith_microblock_stream` in `BlocksInvData` are only ever used in tests, and can be decorated with `#[cfg(test)]`.  Other than that, there's not much overlap.

Reply:
Fixed in daa9099